frontends/web/src/routes/buy/exchange.tsx

```diff
@@ -79,34 +76,28 @@ export const Exchange = ({ code, accounts }: TProps) => {
 
   const hasOnlyOneSupportedExchange = allExchangeDeals ? allExchangeDeals.exchanges.filter(exchange => exchange.supported).length === 1 : false;
 
-  // link locale detection to regionList to having it happen only once page load.
-  // can't use `useLoad` because `detect` function returns void.
+  // update region Select component when `regionList` or `config` gets populated.
   useEffect(() => {
-    languageFromConfig.detect((locale: string) => setLocale(localeMainLanguage(locale)));
-  }, [regionList]);
-
-  // update region Select component when `regionList` or `locale` gets populated.
-  useEffect(() => {
-    if (!regionList || !locale || !config || !nativeLocale) {
+    if (!regionList || !config) {
       return;
     }
-    const regionNames = new Intl.DisplayNames([locale], { type: 'region' });
+    const regionNames = new Intl.DisplayNames([i18n.language], { type: 'region' });
     const regions = regionList.regions.map(region => ({ value: region.code, text: regionNames.of(region.code) } as TOption));
 
-    regions.sort((a, b) => a.text.localeCompare(b.text, locale));
+    regions.sort((a, b) => a.text.localeCompare(b.text, i18n.language));
     setRegions(regions);
 
     if (config.frontend.selectedExchangeRegion) {
       setSelectedRegion(config.frontend.selectedExchangeRegion);
       return;
     }
 
-    const userRegion = getRegionNameFromLocale(nativeLocale);
+    const userRegion = getRegionNameFromLocale(i18n.language);
```

Comment:
Hey there! Sorry, was unaware of this PR before it gets merged. Just a question, if I'm not mistaken `i18n.language` is the language of the app (whatever the user choses), right?

if so, when getting `userRegion, won't it be misleading if we detect it solely from their language? E.g costa rica user might have "Spain" as pre-selected country due to "Spanish" being selected as their language..? Or maybe Im missing something here? :)

> A bug in the locale detection caused the exchange selection flow to break if an empty native locale values was retrieved.

I "catch" a condition when nativeLocale isn't in the list (if nativeLocale is invalid but NOT empty), it'll default to  empty string: `setSelectedRegion(regionAvailable ? userRegion : '');`, which in the UI it'll select the "select region".


Since im unaware that native locale could be empty, this wasn't "caught". So, would it be possible to "catch" this and do similar thing like the above? Im not sure what nativeLocale returned that you said it's empty (empty string maybe?). If so, could we gracefully handle it maybe like so: `const userRegion = getRegionNameFromLocale(nativeLocale || '');` and remove `nativeLocale` check at the top of `useEffect`?

_Note: I haven't experienced an empty native-locale before, so im also a bit curious on what were the steps to reproduce it.._

Reply:
I see what you mean, maybe we should just use i18n.language or config.backend.userLang as a fallback then.

> if so, when getting `userRegion, won't it be misleading if we detect it solely from their language? E.g costa rica user might have "Spain" as pre-selected country due to "Spanish" being selected as their language..? Or maybe Im missing something here? :)

Just assume navite-locale can return anything, also empty string or weird unknown code.

> Note: I haven't experienced an empty native-locale before, so im also a bit curious on what were the steps to reproduce it..

@shonsirsha could you open a new PR and add the logic of guessing the country again, but assume it can be any string.

Reply:
> Just assume navite-locale can return anything, also empty string or weird unknown code.

> @shonsirsha could you open a new PR and add the logic of guessing the country again, but assume it can be any string.

I believe it was already doing this, but we just didn't anticipate that native-locale could be empty, so I believe it crashed because of the check on top of useEffect (`!nativeLocale`). I'll create a PR that will accommodate this.